src/osd/PeeringState.cc

```diff
@@ -1607,22 +1607,21 @@ bool PeeringState::recoverable_and_ge_min_size(const vector<int> &want) const
           pool.info.is_erasure() ? shard_id_t(i) : shard_id_t::NO_SHARD));
     }
   }
-  // We go incomplete if below min_size for ec_pools since backfill
-  // does not currently maintain rollbackability
-  // Otherwise, we will go "peered", but not "active"
-  if (num_want_acting < pool.info.min_size &&
-      (pool.info.is_erasure() ||
-       !cct->_conf->osd_allow_recovery_below_min_size)) {
-    psdout(10) << __func__ << " failed, below min size" << dendl;
-    return false;
-  }
 
   /* Check whether we have enough acting shards to later perform recovery */
-  if (!missing_loc.get_recoverable_predicate()(have)) {
-    psdout(10) << __func__ << " failed, not recoverable" << dendl;
-    return false;
+  if (num_want_acting < pool.info.min_size) {
```

Comment:
@liuchang0812 sorry for not bringing this up earlier, but I think we want to check for `SERVER_OCTOPUS` here to use the new code path and we should revive the old code if this condition does not hold true. This is because in mixed cluster scenarios(for example nautilus and octopus), inconsistent `choose_acting()` behavior can lead to a loop, changing pg_tmp back and forth.

Reply:
alternatively, we might want to condition this on the OSDMap's require_osd_release. Is there potential for problems if we have a smaller acting set with octopus features, and then later a larger/different acting set that doesn't?

Reply:
I'm not sure, do you see a problem in using `SERVER_OCTOPUS`? either way, we should avoid getting into a situation like https://tracker.ceph.com/issues/39441

Reply:
I'm wondering if it's possible to get into a situation where there is disagreement in the PastIntervals as far as whether a particular interval is maybe_rw or not... if differnet OSDs disagreed about that it could cause some weirdness.  But I think this PR isn't changing the active criteria?  In which case I think it's fine...